### ISSUE TITLE
Add forwarded port in the Vagrantfile

### ISSUE THREAD

**Issue by imdhruvgupta** [mit-cml/appinventor-sources#2063]
**Describe the bug**
Port 9876 should be added as a forwarded port in the Vagrantfile so that hot reloading can done as mentioned in the [readme](https://github.com/mit-cml/appinventor-sources#hot-reloading-gwt-code-with-super-dev-mode) .
Port 9900 is [present](https://github.com/mit-cml/appinventor-sources/blob/master/Vagrantfile#L28) as a forwarded port which I think is not used anywhere and should be replaced with 9876.

**Affects**

- [ ] Designer
- [ ] Blocks editor
- [ ] Companion
- [ ] Compiled apps
- [x] Buildserver
- [ ] Other... (please describe)

**Expected behaviour**
http://localhost:9876 should be reachable from the host machine and should redirect the request to VM at the port 9876.

**Steps to reproduce**
Try reaching http://localhost:9876 using any browser after starting the server from the VM.

**Comment by imdhruvgupta** [comment]
@ewpatton Please check. If you agree, i will open the pull request.

**Comment by ewpatton** [comment]
Port 9990 is used for the buildserver. You can go ahead and add port 9876 for the SDM code server.

**Comment by imdhruvgupta** [comment]
@ewpatton I have added port 9876 in the Vagrantfile. Still not able to reach the http://localhost:9876 .
I tried using curl and got the error `curl: (56) Recv failure: Connection reset by peer`. 
I have looked at sources and found that it might solved by binding the Address 0.0.0.0 instead of the default 127.0.0.1 . 
Can you please guide how I should approach that or do you suspect any other solution?

**Comment by ewpatton** [comment]
@imdhruvgupta Are you actually running the code server in the VM? It doesn't start by default so you will need to run `ant devmode` from within a `vagrant ssh` session.

**Comment by imdhruvgupta** [comment]
> @imdhruvgupta Are you actually running the code server in the VM? It doesn't start by default so you will need to run `ant devmode` from within a `vagrant ssh` session.

@ewpatton Yes, I am doing that. I even made curl request from within `vagrant ssh` and got the webpage content for `http://localhost:9876`. But it failed when i did the same from the host.

**Comment by ewpatton** [comment]
@imdhruvgupta I believe that we will need to modify the devmode target so that it binds to 0.0.0.0 rather than 127.0.0.1. The issue right now is that the vagrant port forward only forwards requests to the network interface, not to the machine's loopback interface. By default the SDM server only listens on loopback.

**Comment by imdhruvgupta** [comment]
> @imdhruvgupta I believe that we will need to modify the devmode target so that it binds to 0.0.0.0 rather than 127.0.0.1. 

@ewpatton Yes, Totally agreed. That's what I meant when I said -
> I have looked at sources and found that it might solved by binding the Address 0.0.0.0 instead of the default 127.0.0.1 .

I have searched and found that it can be done by passing `bindAddress` as a parameter with 0.0.0.0 as value but got the same result. How can we modify it?

**Comment by ewpatton** [comment]
The invocation is defined in appengine/build.xml in the devmode target there. You just need to change the arg line passed to the server.

**Comment by imdhruvgupta** [comment]
Thanks @ewpatton for helping out. Please review, I have opened a PR.

**Comment by ewpatton** [comment]
Fixed by #2072
